src/components/FormBuilder/FormBuilder.tsx

```diff
@@ -1,17 +1,129 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
+
 import ConfigProps from "src/interfaces/config-props";
+import ValidationsProps from "src/interfaces/config-validations";
+import InputConfigsProps from "src/types/InputConfigs";
+
+import Button from "../Button/Button";
 import Input from "../Input/Input";
+import PasswordInput from "../PasswordInput/PasswordInput";
+
+import * as util from "src/util";
+import ErrorBlock from "../ErrorBlock/ErrorBlock";
 
 interface FormBuilderProps {
   config: ConfigProps[];
+  setInputValues?: Function;
+  aboveLink?: React.ReactNode;
 }
 
-const FormBuilder: React.FC<FormBuilderProps> = ({ config }) => {
+const FormBuilder: React.FC<FormBuilderProps> = ({
+  config,
+  setInputValues,
+  aboveLink,
+}) => {
+  const [itemProperties, setItemProperties] = useState<InputConfigsProps[]>([]);
+
+  useEffect(() => {
+    const newItemProperties: InputConfigsProps[] = config.map(
+      (formElement) => ({ ...formElement, value: "", validError: "" })
+    );
+    setItemProperties(newItemProperties);
+  }, []);
+
+  const setValue = (e: React.ChangeEvent<HTMLInputElement>, index: number) => {
```

Comment:
Please move validation logic to custom hook if it is possible

Reply:
Done.